### PR TITLE
Rlcs

### PR DESCRIPTION
**Pull request by Carolx715**
Started the code to process data from the RLCS source

---
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54)

### CODE REVIEW

**Review by KavinSatheeskumar — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54)* status: 0 of 2 files reviewed, 15 unresolved discussions (waiting on @Carolx715 and @KavinSatheeskumar)

___
*[`sources/rlcs/main.py` line 9 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1uaaVI9CrwO09ly0Uv:-N1uaaVI9CrwO09ly0Uw:b50sgfk) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L9)):*
> ```Python
> import time
> ```

One thing worth noting is that we have this tool called "format.sh" (in the tools folder), if you go and click that twice, it will automatically format all the python code in the repo.
___
*[`sources/rlcs/main.py` line 10 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1uYiLu6Lv__N-o452F:-N1uYiLvDQty342i2gc3:bk4zn5o) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L10)):*
> ```Python
> 
> 
> msg_index = [
> ```

It might be nice to move this out to a separate file called "rlcs_message_config.py" or something, and then we import this as a constant
___
*[`sources/rlcs/main.py` line 22 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1uZ60H0spMj2FlcW1c:-N1uZ60H0spMj2FlcW1d:bs4w9rr) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L22)):*
> ```Python
> 
> 
> def parse_rlcs(line):
> ```

Might be good to move this function out to RLCS.py, if you already have another file.
___
*[`sources/rlcs/main.py` line 26 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1uZJE35yPiM2cdFJ2B:-N1uZJE35yPiM2cdFJ2C:bbxul9w) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L26)):*
> ```Python
>     # timestamp and msg_type
>     res["msg_type"] = "rlcs"
>     res["timestamp"] = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time()))
> ```

Just noting, make sure that you confirm with Jack that this is the right time format, IDK off the top of my head.
___
*[`sources/rlcs/main.py` line 37 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1uZVAr9LLNriFz61vg:-N1uZVAr9LLNriFz61vh:b-avc3je) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L37)):*
> ```Python
>     if port == "-":
>         return input
>     s = serial.Serial(port, 9600) # this is just copied from parsley, change if needed
> ```

Yeah, we'll get you the actual values, soon
___
*[`sources/rlcs/main.py` line 44 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1uZcc80c_WBROW7M6f:-N1uZcc80c_WBROW7M6g:bn2ab8t) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L44)):*
> ```Python
> 
> 
> def generate_line():
> ```

Might be good to move this to a test file, so we can make good unit tests and stuff.

Side note, it would be wonderful if you could write some unit tests for this code. I can walk you through that process in a bit.
___
*[`sources/rlcs/main.py` line 73 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1u_2lT1CtdOx9lTwfi:-N1u_2lT1CtdOx9lTwfj:b4hwo5f) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L73)):*
> ```Python
> 
>     readline = reader(args.port)
>     # parser = parsley.parse_logger if args.format == 'logger' else parsley.parse_usb_debug
> ```

you had ought be able to remove this comment.
___
*[`sources/rlcs/main.py` line 80 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1u_KhODF7EcbDyHlGh:-N1u_KhODF7EcbDyHlGi:bvp44jx) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L80)):*
> ```Python
>     while True:
>         line = readline()
>         # line = generate_line()
> ```

pls delete :pray:
___
*[`sources/rlcs/main.py` line 87 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1u_P9r0eTiHfOxfY73:-N1u_P9s3crb_La0oiqP:b-ohavt) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L87)):*
> ```Python
>         if line[0] != "W" or line[len(line)-1] != "R":
>             print("Data " + line + " is invalid (must end with R and begin with W")
>             time.sleep(5)
> ```

avoid time.sleep(), it'll cause our code to stop for 5 seconds = a ton of data we just lost
___
*[`sources/rlcs/main.py` line 88 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1u_Vkl8BWU9uSfdesL:-N1u_VkmBnQ9sIesw24R:b33nv4v) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L88)):*
> ```Python
>             print("Data " + line + " is invalid (must end with R and begin with W")
>             time.sleep(5)
>             continue # do we want to throw or use continue?
> ```

I think that we definitively want to use continue, its quite likely we will have messed up data passed in, its best to just toss it and keep moving forward with the rest of the message. Remove the comment.
___
*[`sources/rlcs/main.py` line 90 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1u_j_y2f3aYNq2sjmU:-N1u_j_y2f3aYNq2sjmV:b40l9zd) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L90)):*
> ```Python
>             continue # do we want to throw or use continue?
>         
>         if len(line) > 34 or len(line) < 34:
> ```

if len(line) != 34
___
*[`sources/rlcs/main.py` line 93 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1u_qDm8oPJc7xQ0EJi:-N1u_qDm8oPJc7xQ0EJj:bi6roju) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L93)):*
> ```Python
>             print("Warning: Format {} is wrong".format(line))
>             time.sleep(5)
>             continue # far future possibly
> ```

Make the comment a tad more specific. "In the future, we may try and tease some information out of the message despite its poor formatting"
___
*[`sources/rlcs/main.py` line 95 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1u_xpW6vhSR4Fro1tT:-N1u_xpW6vhSR4Fro1tU:blrbv7s) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L95)):*
> ```Python
>             continue # far future possibly
>         
>         if len(line) == 34:
> ```

remove this if-statement, not needed due to above if statement
___
*[`sources/rlcs/main.py` line 99 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1uaEy1BfK1-q99CY2_:-N1uaEy1BfK1-q99CY2a:b-j3p0yo) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L99)):*
> ```Python
>             parsed_data = parse_rlcs(stripped_data)
> 
>         print(rlcs.fmt_line(parsed_data)) # print debug code 
> ```

actually, if args.solo is true, you would want to print this anyways, care to move this line down where you call "sender.send" and put it in an else block after that if statement?
___
*[`sources/rlcs/main.py` line 100 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1ua4xMDLZhgRkm1Cy9:-N1ua4xMDLZhgRkm1CyA:b-so594p) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L100)):*
> ```Python
> 
>         print(rlcs.fmt_line(parsed_data)) # print debug code 
>         time.sleep(5)
> ```

avoid time.sleep()

**Comment by Carolx715** [comment]
Pushed most of the changes, will report back on the time format used once I get the info

**Review by KavinSatheeskumar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @Carolx715 and @KavinSatheeskumar)

___
*[`sources/rlcs/main.py` line 37 at r1](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1uZVAr9LLNriFz61vg:-N1y-qgf-iX9l53z1BPa:boonq6i) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/503f713a857548ad99ae4788649f9c2a5010f52f/sources/rlcs/main.py#L37)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Yeah, we'll get you the actual values, soon
</blockquote></details>

115200 instead of 9600
___
*[`sources/rlcs/rlcs_test.py` line 7 at r3](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1y05oSC3X-E9iksujR:-N1y05oSC3X-E9iksujS:b-igun70) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/5eacf24fdea282f7890d94f184128a6f89c93db5/sources/rlcs/rlcs_test.py#L7)):*
> ```Python
> from random import random
> ```

Look at "omnibus/sources/parsley/parsley_test.py" to get examples of tests (you probably won't need as many as are in that file, but the format is what you should be after).

You essentially have to make a class with a bunch of methods. Each one of those is a test that is run.

**Review by KavinSatheeskumar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54)* status: 0 of 4 files reviewed, 4 unresolved discussions (waiting on @Carolx715 and @KavinSatheeskumar)

___
*[`sources/rlcs/main.py` line 22 at r5](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2Dy-dc-A7nMwr0ocVe:-N2Dy-dc-A7nMwr0ocVf:bdonzbl) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/f39678734d409737739ab59b0517063a810c0906/sources/rlcs/main.py#L22)):*
> ```Python
>     parser = argparse.ArgumentParser()
>     parser.add_argument('port', help='the serial port to read from, or - for stdin')
>     parser.add_argument('--format', default='usb',
> ```

Is this actual used? If not, it had ought be removed.
___
*[`sources/rlcs/main.py` line 42 at r5](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2Dy8Sd9raLOZZzFWuE:-N2Dy8Sd9raLOZZzFWuF:b-hngdm4) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/f39678734d409737739ab59b0517063a810c0906/sources/rlcs/main.py#L42)):*
> ```Python
>             continue
> 
>         parsed_data = rlcs.parse_rlcs(line)
> ```

Jack Suggested that you make your parse_rlcs function do the data checking as well. Instead of first checking then parsing. 

so your parse_rlcs will return "None" if the parse fails. You store this to the value parsed_data and do the below. This can also help with testing. 



_Code snippet:_
```Python
if not parsed_data:
    continue
```
___
*[`sources/rlcs/rlcs.py` line 35 at r5](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2Dyk-P-QUoDZXr_nuz:-N2Dyk-P-QUoDZXr_nv-:b-9lc0uc) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/f39678734d409737739ab59b0517063a810c0906/sources/rlcs/rlcs.py#L35)):*
> ```Python
>         print("Data " + line + " is invalid (must end with R and begin with W")
> 
>     if len(line) != 34:
> ```

This is my bad, but apparently the number of items might vary. So instead of 34, it had ought be 2 + len(msg_index).
___
*[`sources/rlcs/rlcs_test.py` line 7 at r3](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N1y05oSC3X-E9iksujR:-N2DxqzK9j3rt-XKGjwE:b9o3jvi) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/5eacf24fdea282f7890d94f184128a6f89c93db5/sources/rlcs/rlcs_test.py#L7)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

Look at "omnibus/sources/parsley/parsley_test.py" to get examples of tests (you probably won't need as many as are in that file, but the format is what you should be after).

You essentially have to make a class with a bunch of methods. Each one of those is a test that is run.
</blockquote></details>

One extra test where you just pass in hot garbage would be nice.
Also, in keeping with Jack's comment above, I suggest the following change to testing. 

Instead of just having bool "True" "False" answers, have the actual final object that needs to be produced.

**Review by BluCodeGH — CHANGES_REQUESTED** [review]
This is looking great, I'm just being nitpicky since omnibus is my baby.

> Reviewed 1 of 4 files at r5, 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-:-N2XISQIE0XmWa7wwEgQ:b-7o60kt)* status: all files reviewed, 14 unresolved discussions (waiting on @Carolx715)

___
*[`sources/rlcs/main.py` line 46 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XDyOY0Q8o55UKRJvH:-N2XDyOY0Q8o55UKRJvI:b-thgxnu) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/main.py#L46)):*
> ```Python
>             sender.send(CHANNEL, parsed_data)
>         else:  # print debug code to console
>             print(rlcs.fmt_line(parsed_data))
> ```

Probably makes sense to print whether or not omnibus is connected.
___
*[`sources/rlcs/rlcs.py` line 26 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XFqf540tC-UaKdLL6:-N2XFqf540tC-UaKdLL7:b91jvnp) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs.py#L26)):*
> ```Python
>     # timestamp and msg_type
>     res["msg_type"] = "rlcs"
>     res["timestamp"] = time.time()
> ```

The reason parsley includes these fields is because they are part of the CAN message data that needs to be relayed. You don't need to worry about that, so easier to just return the data dictionary. Omnibus itself will keep track of the timestamp of when you sent the message. Make sure to update `fmt_line` to reflect the change as well.
___
*[`sources/rlcs/rlcs.py` line 34 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XI9lJ84qwAX-SQnYP:-N2XI9lJ84qwAX-SQnYQ:b-jr8a3b) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs.py#L34)):*
> ```Python
> 
> 
> def check_data_is_valid(line):
> ```

Could use a comment saying what valid data looks like.
___
*[`sources/rlcs/rlcs.py` line 39 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XGEkeD818bBO6hJdp:-N2XGEkeD818bBO6hJdq:b-qh1aoz) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs.py#L39)):*
> ```Python
>     if line[0] != "W" or line[len(line)-1] != "R":
>         is_valid = False
>         print("Data " + line + " is invalid (must end with R and begin with W")
> ```

Please make your error/warning/etc messages consistent. Right now you have one with no heading, one warning, and one error, but in all cases they mean the parsing failed since something was wrong with the data. I'd probably go for warning in all cases, and have all of them print the entire invalid `line` to aid in debugging.
___
*[`sources/rlcs/rlcs.py` line 41 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XEqpB-a_QWof6efTk:-N2XEqpB-a_QWof6efTl:b-3evmhr) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs.py#L41)):*
> ```Python
>         print("Data " + line + " is invalid (must end with R and begin with W")
> 
>     if len(line) != 4*len(msg_index)+2:
> ```

You should perform the length check before accessing any characters in `line`, since otherwise you'll crash if passed an empty line.
___
*[`sources/rlcs/rlcs.py` line 47 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XF1fEB6FYC0VeFTjX:-N2XF1fEB6FYC0VeFTjY:b-ra880r) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs.py#L47)):*
> ```Python
>         # In the future, we may want to extract information from the message despite poor formatting
> 
>     for i in range(1, len(line)-1):
> ```

You can iterate through characters directly: `for c in line:`. You won't get the index in the error message but I'd rephrase the error to be something like `Error: Invalid hex data {line}`.
___
*[`sources/rlcs/rlcs.py` line 48 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XFJbM0Up7cxG2eemE:-N2XFJbM0Up7cxG2eemF:b-34mt6h) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs.py#L48)):*
> ```Python
> 
>     for i in range(1, len(line)-1):
>         if line[i] not in "abcdef0123456789":
> ```

The hex you'll be getting is uppercase, but probably best to support both upper and lower case.
___
*[`sources/rlcs/rlcs_test.py` line 45 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XH4Pb8qsWlKWnNlHq:-N2XH4Pb8qsWlKWnNlHr:bj5bztq) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs_test.py#L45)):*
> ```Python
> 
>         for _ in range(8):
>             hexnum = hex(int(random()*65536))[2:6]
> ```

It's probably easier to generate a random string directly, eg `''.join(random.choices("ABCDEF0123456789", k=4*len(msg_index)))`
___
*[`sources/rlcs/rlcs_test.py` line 53 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XHlfM1X7FZFJMx4td:-N2XHlfM1X7FZFJMx4te:b-hfpv1m) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs_test.py#L53)):*
> ```Python
> 
> 
> if __name__ == "__main__":
> ```

remove?
___
*[`sources/rlcs/rlcs_message_config.py` line 1 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XEBLP79kddaMcNECP:-N2XEBLP79kddaMcNECQ:bivyijz) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs_message_config.py#L1)):*
> ```Python
> msg_index = [
> ```

I would recommend naming this file `config.py` for consistency with the other sources/sinks. You should also add a comment explaining what this variable configures, and constants like this should be `ALL_CAPS`.

**Review by BluCodeGH — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54)* status: all files reviewed, 2 unresolved discussions (waiting on @Carolx715)

___
*[`sources/rlcs/rlcs.py` line 26 at r6](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2XFqf540tC-UaKdLL6:-N2gCWSw3tCW4oTucssS:b-ayd08y) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/62577d314e96f337a94cb65816806c81256ea4b7/sources/rlcs/rlcs.py#L26)):*
<details><summary><i>Previously, BluCodeGH (BluCode) wrote…</i></summary><blockquote>

The reason parsley includes these fields is because they are part of the CAN message data that needs to be relayed. You don't need to worry about that, so easier to just return the data dictionary. Omnibus itself will keep track of the timestamp of when you sent the message. Make sure to update `fmt_line` to reflect the change as well.
</blockquote></details>

You don't need `msg_type` either, that's defined based on the channel you send the message to in `main.py`. I think you'll just want `res = {}` and to replace all references of `res["data"]` with just `res`.
___
*[`sources/rlcs/rlcs_test.py` line 7 at r7](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2gDHUY1_uT5ttHocWY:-N2gDHUZEFXjlHr24JX8:b6ir0hj) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/42f0755c058dcba42f5331d6dfb1054f684ee1f8/sources/rlcs/rlcs_test.py#L7)):*
> ```Python
> import random
> import rlcs.rlcs as rlcs
> from rlcs.config import MSG_INDEX
> ```

Why are your imports like this? Does it not work to `import rlcs` and `from config import MSG_INDEX` (and then you don't need the __init__.py)?

**Comment by Carolx715** [comment]
> Why are your imports like this? Does it not work to import rlcs and from config import MSG_INDEX (and then you don't need the init.py)?

Re: this, pytest mistakes `import config.py` for `ni/config.py` instead of `rlcs/config.py` and the only way I found around that was to make RLCS a package. Is there a better way around this?

**Review by BluCodeGH — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54)* status: all files reviewed, 2 unresolved discussions (waiting on @Carolx715)

___
*[`sources/rlcs/rlcs.py` line 6 at r8](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2hCXJx-Kg--gETzxZP:-N2hCXJx-Kg--gETzxZQ:bd5ty59) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/47fcbef610b448935f644d09b913f35f5bf429fc/sources/rlcs/rlcs.py#L6)):*
> ```Python
> def fmt_line(parsed_data):
>     msg_type = parsed_data['msg_type']
>     data = parsed_data["data"]
> ```

:eyes: 
___
*[`sources/rlcs/rlcs_test.py` line 7 at r7](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2gDHUY1_uT5ttHocWY:-N2hCaa49SPEb_u_PBC1:bv6qjrn) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/42f0755c058dcba42f5331d6dfb1054f684ee1f8/sources/rlcs/rlcs_test.py#L7)):*
<details><summary><i>Previously, BluCodeGH (BluCode) wrote…</i></summary><blockquote>

Why are your imports like this? Does it not work to `import rlcs` and `from config import MSG_INDEX` (and then you don't need the __init__.py)?
</blockquote></details>

Ugh that's awful. I think renaming `config.py` to something like `config_rlcs.py`, while ugly, is the lesser of two evils here.

**Review by Carolx715 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54)* status: 1 of 6 files reviewed, 2 unresolved discussions (waiting on @BluCodeGH and @Carolx715)

___
*[`sources/rlcs/rlcs.py` line 6 at r8](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2hCXJx-Kg--gETzxZP:-N2hIyhk2leuTFOJLixZ:b-irq5c7) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/47fcbef610b448935f644d09b913f35f5bf429fc/sources/rlcs/rlcs.py#L6)):*
<details><summary><i>Previously, BluCodeGH (BluCode) wrote…</i></summary><blockquote>

:eyes: 
</blockquote></details>

whoops

**Review by BluCodeGH — CHANGES_REQUESTED** [review]
> Reviewed 5 of 5 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54)* status: all files reviewed, 1 unresolved discussion (waiting on @Carolx715)

___
*[`sources/rlcs/rlcs.py` line 6 at r8](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2hCXJx-Kg--gETzxZP:-N2hJlVyDqNZYBBlbeYx:ba3w5ly) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/47fcbef610b448935f644d09b913f35f5bf429fc/sources/rlcs/rlcs.py#L6)):*
<details><summary><i>Previously, Carolx715 (Carol Xu) wrote…</i></summary><blockquote>

whoops
</blockquote></details>


Please test your code.

**Comment by Carolx715** [comment]
*[`sources/rlcs/rlcs.py` line 6 at r8](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2hCXJx-Kg--gETzxZP:-N2hX2Zv0HrdSEBzgW98:beisn4e) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/47fcbef610b448935f644d09b913f35f5bf429fc/sources/rlcs/rlcs.py#L6)):*
<details><summary><i>Previously, BluCodeGH (BluCode) wrote…</i></summary><blockquote>


Please test your code.
</blockquote></details>

sorry bout that, fixed now. Also added another test to see if it returns a value, lmk if that's ok

**Review by BluCodeGH — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-:-N2hf3-QEXGB6m_QvRvA:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @Carolx715)

___
*[`sources/rlcs/rlcs.py` line 6 at r8](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-N2hCXJx-Kg--gETzxZP:-N2hf-pC1jYSczDXvgPd:b-7m6032) ([raw file](https://github.com/waterloo-rocketry/omnibus/blob/47fcbef610b448935f644d09b913f35f5bf429fc/sources/rlcs/rlcs.py#L6)):*
<details><summary><i>Previously, Carolx715 (Carol Xu) wrote…</i></summary><blockquote>

sorry bout that, fixed now. Also added another test to see if it returns a value, lmk if that's ok
</blockquote></details>

That's alright, sounds good.

**Review by KavinSatheeskumar — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @Carolx715)

**Review by KavinSatheeskumar — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/omnibus/54#-:-N2hfTqv-52yHJtCYDUg:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @Carolx715)
